app/script/calling/entities/FlowEntity.js

```diff
@@ -856,24 +855,47 @@ z.calling.entities.FlowEntity = class FlowEntity {
 
         const isModeDefault = this.negotiationMode() === z.calling.enum.SDP_NEGOTIATION_MODE.DEFAULT;
         if (isModeDefault && sendingOnTimeout) {
-          if (!this._containsRelayCandidate(iceCandidates)) {
-            const logMessage = `No relay ICE candidates in local SDP. Timeout reset\n${iceCandidates}`;
-            this.callLogger.warn(logMessage, iceCandidates);
-            return this._setSendSdpTimeout(false);
+          const MIN_NUMBER_OF_RELAY_SERVERS = 2;
+          if (this._getNumberOfRelayCandidates(iceCandidates) < MIN_NUMBER_OF_RELAY_SERVERS) {
+            const logMessage = `Less than ${MIN_NUMBER_OF_RELAY_SERVERS} relay ICE candidates in local SDP. Timeout reset\n${iceCandidates}`;
+            this.callLogger.warn(logMessage);
+            return this._setSendSdpTimeout();
           }
         }
 
+        const iceCandidateTypes = iceCandidates.reduce((types, candidateStr) => {
+          const typeMatches = candidateStr.match(/typ (\w+)/);
+          if (!typeMatches) {
+            return types;
+          }
+          const candidateType = typeMatches[1];
+          types[candidateType] = types[candidateType] ? types[candidateType] + 1 : 1;
```

Comment:
No need for a ternary:

`types[candidateType] = types[candidateType] + 1 || 1;` should give you the same result as `underfined + 1` is `NaN` with is not truthy.

Reply:
Ok works, but feels a little cryptic...